### ISSUE TITLE
UniversalDetector.done does not work

### ISSUE THREAD

**Issue by kovidgoyal** [chardet/chardet#126]
Trying the example from the docs:

```sh
python2 -c "import urllib; rawdata = urllib.urlopen('http://yahoo.co.jp/').read(); from chardet.universaldetector import UniversalDetector; detector = UniversalDetector(); detector.feed(rawdata); print detector.done"
```
prints False

No matter how large a document is fed to UniversalDetector it never gets done.

**Comment by dan-blanchard** [comment]
You have to call `detector.close()` to get the final prediction when you're done feeding it data. The only times `detector.done` would be `True` without that would be if we determined the encoding with an extremely high confidence.

**Comment by sigmavirus24** [comment]
@kovidgoyal `UniversalDetector.feed` *may* be able to make a decision if there's enough information and in that case *may* set `self.done = True` but as @dan-blanchard says, the most reliable way is to call `UniversalDetector.close` when you know you've provided all the data possible.

**Comment by kovidgoyal** [comment]
Yeah I saw that from reading the source. I just figured it would be far more likely to reach the threshold than it actually is. From what I can see, it basically is there only for BOM based detection. I was not able to trigger it in any other cases, with documents of various lengths in various single-byte encodings.  Maybe the documentation should have a line stating that it is largely useless with the exception of BOM based decoding. It saves people from going through the effort of setting up incremental feeding in the hopes of a performance gain.
